projects/components/src/utils/test/widget-object/angular-widget-object.ts

```diff
@@ -162,6 +162,37 @@ export class TestElement implements Iterable<TestElement> {
         return this.elements.map((el) => new TestElement([el], this.fixture));
     }
 
+    /**
+     * Send a keyboard event of type {@param eventType} with properties {@param eventProperties} on this element.
+     * Setting the event properties is done with `Object.defineProperty` on the created event. This allows setting
+     * properties like `which` that is deprecated and cannot be set with the native approach of creating keyboard event.
+     * @param eventType The keyboard event type like 'keyup', 'keydown', 'keypress'
+     * @param eventProperties properties of the event like `code`, `key` etc.
+     */
+    sendKeyboardEvent(eventType: string, eventProperties: { [name: string]: unknown }): void {
+        const element = this.elements[0].nativeElement as HTMLElement;
+        const event = new KeyboardEvent(eventType, { bubbles: true });
+        Object.keys(eventProperties).forEach((key) => {
+            Object.defineProperty(event, key, { value: eventProperties[key] });
+        });
+        element.dispatchEvent(event);
+        this.detectChanges();
+    }
+
+    /**
+     * Returns classes of first element as a string array
+     */
+    classes(): string[] {
+        return Object.keys(this.elements[0].classes);
+    }
+
+    /**
+     * Returns the parent of first element
+     */
+    parent(): TestElement {
```

Comment:
This seems catered to a very specific example and we are using it for reaching outside of the widget object. Let's talk about how we can improve this